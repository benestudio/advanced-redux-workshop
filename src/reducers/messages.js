export default (state = {}, action) => {
  switch (action.type) {
    case 'MESSAGES_FETCH_START':
      return {
        isFetching: true,
        data: [],
      };
    case 'MESSAGES_FETCH_SUCCESS':
      return {
        isFetching: false,
        data: action.payload.data,
      };
    default:
      return state;
  }
}
