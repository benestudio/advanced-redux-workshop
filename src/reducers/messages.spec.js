import messages from './messages';

describe('messages reducer', () => {
  it('is a function', () => {
    expect(messages.constructor).toBe(Function);
  })

  it('should mark fetch start correctly', () => {
    const action = { type: 'MESSAGES_FETCH_START' };
    expect(messages({}, action)).toMatchSnapshot();
  })
  it('should output messages on fetch success', () => {
    const data = [
      {
        id: '1',
        sender: 'Alice',
        message: 'Hi Bob!',
        created: '1532432201308',
      },
      {
        id: '2',
        sender: 'Bob',
        message: 'Hi Alice!',
        created: '1532432212345',
      },
    ];
    const action = {
      type: 'MESSAGES_FETCH_SUCCESS',
      payload: {
        data,
      },
    };
    expect(messages({}, action)).toMatchSnapshot();
  })
});
