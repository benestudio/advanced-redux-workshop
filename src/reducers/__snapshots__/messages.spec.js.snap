// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages reducer should mark fetch start correctly 1`] = `
Object {
  "data": Array [],
  "isFetching": true,
}
`;

exports[`messages reducer should output messages on fetch success 1`] = `
Object {
  "data": Array [
    Object {
      "created": "1532432201308",
      "id": "1",
      "message": "Hi Bob!",
      "sender": "Alice",
    },
    Object {
      "created": "1532432212345",
      "id": "2",
      "message": "Hi Alice!",
      "sender": "Bob",
    },
  ],
  "isFetching": false,
}
`;
